### ISSUE TITLE
func-call-spacing never with allowNewlines

### ISSUE THREAD

**Issue by MNITD** [eslint/eslint#14266]
**What rule do you want to change?**
[func-call-spacing](https://eslint.org/docs/rules/func-call-spacing)

**Does this change cause the rule to produce more or fewer warnings?**
Fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
Either allow using `allowNewlines` option for both `always`/`never` or add a new option that will allow using function parentheses on the new line but forbid usage of space after function identifier

**Please provide some example code that this change will affect:**

```js
test
  .meta(owners) // <--  ESLint: Unexpected whitespace between function name and paren.(func-call-spacing)
  ('Login with form', async (t) => { 
    await loginWithForm(t);
    await checkDashboardLayout(t);
  })
  .after (saveLogs); // <-- ESLint: Unexpected whitespace between function name and paren.(func-call-spacing)
```

**What does the rule currently do for this code?**
Currently, there is no ability to use `allowNewlines` option together with `never` so the only way if I want to have no space after function identifier is 
```
"func-call-spacing": ["error", "never"]
```

**What will the rule do after it's changed?**
What I want is to have the ability to use eslint rule in the following way:
```
func-call-spacing: ["error", "never", { "allowNewlines": true }]
```
then the code mentioned above should have only one error:
```js
test
  .meta(owners) // <-- Now, it's okay, no warnings here
  ('Login with form', async (t) => { 
    await loginWithForm(t);
    await checkDashboardLayout(t);
  })
  .after (saveLogs); // <-- ESLint: Unexpected whitespace between function name and paren.(func-call-spacing)
```

**Are you willing to submit a pull request to implement this change?**
It depends, if I will get an assistance in the meaning of recommendations on what is a better approach to implement the feature then I am ready to contribute to the  project

**Comment by mdjermanovic** [comment]
Hi @MNITD, thanks for the issue! 

The team has made the difficult decision to freeze stylistic rules (please find more details in [this blog post](https://eslint.org/blog/2020/05/changes-to-rules-policies)), so we won't be making any changes in the `func-call-spacing` rule, except for fixing bugs and adding support for new ECMAScript features. You can always make a custom rule or publish a plugin.

I'm closing this issue per our [rules policies](https://github.com/eslint/eslint#stylistic-rule-updates).
